src/visualizers/widgets/ArtifactIndex/ArtifactIndexWidget.js

```diff
@@ -139,6 +147,31 @@ define([
         }
     };
 
+    ArtifactIndexWidget.prototype.editInPlace = function(event, opts) {
+        const el = $(event.target);
+        const id = opts.nodeId;
+        const attr = opts.targetAttribute;
+
+        el.editInPlace({
+            css: {
+                'z-index' : 1000
+            },
+            onChange: async (oldVal, newVal) => {
+                if (newVal && newVal !== oldVal) {
+                    const confirmed = opts.confirmation ? await opts.confirmation.call(this,
+                        opts.nodeName,
+                        {newValue: newVal, oldValue: oldVal}
```

Comment:
As a minor point, I am not a fan of packing arguments into an options object unless there are a lot of optional arguments. Otherwise, it obscures the method signature without any real benefit.